### PR TITLE
Scoped ids

### PR DESCRIPTION
**Pull request by tim-friedrich**

### CODE REVIEW

**Comment by amatteini** [comment]
In our application, we are using `Minuteman::User[id]` to retrieve exiting users, but with this change we always create a new user instead. Could you clarify on this?

I'm thinking we could now build User#uid by including the scope in it, eg.    
 ```
  def uid
     @scope + ':' + @id
  end
```

It seems we could also remove few 'ohm' requirement from user.rb, and 'securerandom' from scope_model

**Comment by tim-friedrich** [comment]
I agree the the requrements you listed can be removed. 

Due to Minuteman::User[id]:
I i dont miss anything we are just creating new ruby objects here. there is no overhad in redis. (the global id is not involved) it is up to the user to ensure that the userId exists in the given scope.

could you may clarify why you want to use the old uid? (it was used in the past to track model ids within minuteman. i removed it save memory in redis)

**Comment by amatteini** [comment]
I meant that uid and self.[] methods are in use in our source, so we need to change the code accordingly in case we want to modify or remove those.

File: test/minuteman_test.rb [hunk 1]
```diff
@@ -211,3 +212,18 @@
     assert user.count("login:attempts").day.count == 3
   end
 end
+
+# scope "do action throught a user with different scopes" do
```
Comment (amatteini) [inline test/minuteman_test.rb]: Can you please add a note or remove? Is this pending?

Comment (tim-friedrich) [reply test/minuteman_test.rb]: Removed it.

**Pull request merged**